armsrc/Standalone/hf_mfcsim.c

```diff
@@ -0,0 +1,116 @@
+//-----------------------------------------------------------------------------
+// RayCN, 2021
+//
+// This code is licensed to you under the terms of the GNU GPL, version 2 or,
+// at your option, any later version. See the LICENSE.txt file for the text of
+// the license.
+//-----------------------------------------------------------------------------
+// main code for mifare classic simulator aka MFCSIM
+//-----------------------------------------------------------------------------
+#include <inttypes.h>
+#include "ticks.h"
+#include "proxmark3_arm.h"
+#include "BigBuf.h"
+#include "commonutil.h"
+#include "fpgaloader.h"
+#include "util.h"
+#include "dbprint.h"
+#include "spiffs.h"
+#include "standalone.h" // standalone definitions
+#include "appmain.h"
+#include "string.h"
+#include "iso14443a.h"
+#include "mifarecmd.h"
+#include "crc16.h"
+#include "mifaresim.h"  // mifare1ksim
+#include "mifareutil.h"
+
+/*
+ * `hf_mfcsim` simulates mifare classic 1k dumps uploaded to flash.
+ * It requires RDV4 hardware (for flash and battery).
+ *
+ * On entering stand-alone mode, this module will start simulating.
+ * Data is read from bin dump file uploaded to flash memory (hf_mfcsim_dump_1.bin).
+ * Only support mifare classic 1k
+ *
+ * LEDs:
+ * - LED A: initializing
+ * - LED B: simulating
+ * - LED C blinking: data transmiting
+ *
+ * To upload input file (eml format) to flash:
+ * - mem spiffs upload -s <filename> -d hf_mfcsim_dump_1.bin
+ * To delete the input file from flash:
+ * - mem spiffs remove -f hf_mfcsim_dump_1.bin
+ *
+ */
+
+#define HF_MFCSIM_INPUTFILE_SIM         "hf_mfcsim_dump_1.bin"
```

Comment:
and maybe remove the `_1`  part in the file name.  a bit hard to remember

Reply:
I was thinking of doing a follow-up patch to allow a few different slots, though I admittedly haven't looked at how difficult that would be to do.

If there are multiple slots, having them be _1, _2, _3, and _4 would make a degree of sense.

Reply:
that would make some sense.   But for now it doesn't support it.

Reply:
The difficulty is how to indicate which card is currently being simulated.
the function "Mifare1ksim" has used all LEDs.
sadness....

Reply:
you can flash like led the number of times == active slot?